### ISSUE TITLE
Bad value for ExecutablePath in generated Visual Studio property sheet conanbuildinfo.props

### ISSUE THREAD

**Issue by youcefl** [conan-io/conan#998]
Using 0.18.1 I get this value for tag ExecutablePath in generated
conanbuildinfo.props:
`  
<PropertyGroup>
    <ExecutablePath>E:/users/me/.conan/data/pkg1Name/0.1.0/pkgOwner/stable/package/51314285077f9a78ce6c594610720b3410e6f482/bin;E:/users/me/.conan/data/pkg2Name/58.2.0/pkgOwner/stable/package/e0a02d496bbb652b6295152dfce0d3937acc0b56/bin;$(ExecutablePath)</ExecutablePath>
  </PropertyGroup>
`
It should be %(ExecutablePath) not $(ExecutablePath).
Also, is it possible to (on option) have the generated conanbuildinfo.props file
not contain this tag?

**Comment by memsharded** [comment]
Currently there is no way to define optionality in generators besides they taking the conanfile and being able to read the options.

I think it might be possible just to clean the things you don't want from your consumer ``conanfile.py``, something like:

```python
def package_info(self):
    self.deps_cpp_info["pkg1Name"].bindirs = []
    self.deps_cpp_info["pkg2Name"].bindirs = []
```
You could iterate over package names, and you can also use options of your consumer recipe to enable or disable this removal. Please try and tell me.

**Comment by youcefl** [comment]
I thought about overriding the value of bindirs as you propose
but it requires updating all of my packages right?
Also I think there still is a problem with the $(ExecutablePath)
which should be replaced by %(ExecutablePath) am I right?

**Comment by memsharded** [comment]
Yes, yes, that  path might be a problem, lets check it.

Not necessary to change all the recipes, only those that are being opened for edition as consumers. So adding that can be done gradually as you go.

Another option could be just forking the standard VS generator and customizing it for your specific needs, you could use your own environment variables, for example, or even ``conanfile.options`` to activate/deactivate behavior. The current generator is very short and simple.

Note: We are adding multi-config packages, so generators will have to adapt accordingly to use this feature, so it might need some improvements in the future.

**Comment by memsharded** [comment]
This has been fixed in #1017 (the bug about the %), will be released next 0.20.

Did you have the chance to consider the other comments and ideas? Thanks!

**Comment by davidkjaer** [comment]
Hi,

We are using conan.io within our company with different generators - both CMake and Visual Studio and text. Unfortunately the above change seems to have broken a build dependent on a .prop-file created by a Visual Studio generator. More specifically the 'CL.exe' can no longer be found during the build which pretty devastating :). Furthermore, it is to be expected if ExecutablePath is incorrect.

In my opinion the original value (i.e. pre-0.20.0) of the ExecutablePath-element in the generated .prop-file was correct. Since ExecutablePath is a property its value should be referenced by '$' - not '%'.

We have fixed our build by downgrading conan.io to version 0.19.1. This works as a short term solution but is not really sustainable in the long run.

**Comment by memsharded** [comment]
Ups, we are very sorry. Of course, will have a look asap, will be fixed for next release or we might even generate a minor version for you.

**Comment by memsharded** [comment]
I am sorry @youcefl , it seems you were wrong, and I didn't check properly, my fault. @davidkjaer is right, and the correct value ``$(ExecutablePath)`` was already there, so we introduced a regression.
I have reproduced it, fixed and opened a 0.20.2 branch: https://github.com/conan-io/conan/tree/release/0.20.2

You can run from sources. Please report any other issue asap, we'll probably try to release this 0.20.2 tomorrow. Thanks very much for reporting and your help.

**Comment by memsharded** [comment]
Hi @davidkjaer,

While fixing the issue, I decided to review some old issue related to this generator: #264, and I could come with a solution quickly, I have already submitted it for next release 0.21: https://github.com/conan-io/conan/pull/1052/files

It is likely that the visual studio generator could be further improved, if you use it, please give feedback, any help testing and improving it is appreciated.

**Comment by youcefl** [comment]
Sorry then for causing you to introduce a regression, the behavior experienced
by @davidkjaer is the one I had: Visual Studio failed to find CL.exe. Unfortunately
I didn't have time to work on this issue and it is not a blocker for us because
in production we use an older version which does not generate an ExecutablePath
tag in the props file.

**Comment by memsharded** [comment]
No worries @youcefl, these things happen, it was mostly my fault :)
We will be releasing the minor 0.20.2 fixing this tomorrow. Would be great if you could upgrade your systems on a regular basis and provide feedback, until we can release 1.0 and stabilize out of beta.

**Comment by memsharded** [comment]
This has been released in conan 0.20.2: http://docs.conan.io/en/latest/changelog.html
